### PR TITLE
Upload user code on nextjs branch

### PR DESCRIPTION
**Pull request by radu1122**
# Pull Request Template

## Type of change

-   [ ] 🍕 New feature
-   [ ] 🐛 Bug Fix
-   [ ] 🔥 Breaking change
-   [ ] 🧑‍💻 Improvement
-   [ ] 📝 Documentation Update

## Checklist

-   [ ] My code follows the contributor guidelines of this project;
-   [ ] I have updated the documentation;
-   [ ] I have added tests;
-   [ ] New and existing unit tests pass locally with my changes;

### CODE REVIEW

File: src/genezio.ts [hunk 1]
```diff
@@ -168,6 +168,7 @@ Use --backend to deploy only the backend application.`,
             });
             exit(1);
         });
+        process.exit(0);
```
Comment (costinsin) [inline src/genezio.ts:171]: Why is this necessary? Isn't the function returning anyway and exiting the program?

File: src/commands/deploy/command.ts [hunk 2]
```diff
@@ -4,21 +4,56 @@ import { debugLogger } from "../../utils/logging.js";
...
 import { nextJsDeploy } from "./nextjs.js";
+import path from "path";
+import { createTemporaryFolder, zipDirectory } from "../../utils/file.js";
+import { getPresignedURLForProjectCodePush } from "../../requests/getPresignedURLForProjectCodePush.js";
+import { uploadContentToS3 } from "../../requests/uploadContentToS3.js";
+import { YamlConfigurationIOController } from "../../yamlProjectConfiguration/v2.js";
 
 export async function deployCommand(options: GenezioDeployOptions) {
     await interruptLocalProcesses();
+    // create archive of the project
```
Comment (costinsin) [inline src/commands/deploy/command.ts:15]: This comment is not related to the code below it. I would remove it.

**Review by costinsin — APPROVED** [review]

File: src/genezio.ts (hunk 1 above)
Comment (vladiulianbogdan) [reply src/genezio.ts:171]: We have a problem on the build machine. If we launch `genezio deploy` from another process, it seems that the process does not exit and we suspect this is because of the stdin read. We do the `process.exit(0)` as a hack for now to unblock this. We should figure it out why the process does not exit.

**Review by costinsin — CHANGES_REQUESTED** [review]
This PR introduces a bug that will not deploy the Next.js code if the user skips the CDN deployment spinner. This happens because if any key is pressed, `process.exit` will be called, not letting the code upload finish after deployment.

**Comment by costinsin** [comment]
I have fixed the bug mentioned above in https://github.com/Genez-io/genezio/pull/1161.

**Review by costinsin — APPROVED** [review]
